app/controllers/api/v1/comments_controller.rb

```diff
@@ -0,0 +1,28 @@
+class Api::V1::CommentsController < ApplicationController
+  def index
+    user_posts = current_user.posts.includes(:comments)
+    comments_by_post = user_posts.map { |post| { post_id: post.id, comments: post.comments } }
+    render json: { success: true, comments_by_post: comments_by_post }
+  end
+
+  def create
+    post = current_user.posts.find_by(id: params[:post_id])
+    if post
+      comment = post.comments.build(comment_params)
+      comment.user = current_user
+      if comment.save
+        render json: { success: true, message: 'Comment created successfully' }, status: :created
+      else
+        render json: { success: false, message: 'Comment could not be created' }, status: :bad_request
+      end
+    else
+      render json: { success: false, message: 'Post not found' }, status: :not_found
+    end
+  end
+
+  private
+
+  def comment_params
+    params.require(:comment).permit(:Text)
+  end
+end
```

Comment:
Good job done so far, but there are still issues when trying to create a comment for a post using your API endpoints, expected to be configured as a POST action /users/:user_id/posts/:id/comments. Please note that the authorization used while building your project would be different from the authorization while trying to access your API endpoint. You are trying to use the current_user helper, which would require authorization for your API endpoint. Unlike other functionalities you would be required to implement authorization as you are to create an API endpoint to add a comment to a post. `Remember that the owner of the comment is the user that makes it; it should not be set by passing a parameter`. To help you with this I suggest you use the [devise token auth](https://github.com/lynndylanhurley/devise_token_auth) or any other web token library of your choice that helps with API authorization. Please fix this 🙏